datasync/chaindatafetcher/kas/repository_test.go

```diff
@@ -0,0 +1,137 @@
+	funds       = big.NewInt(100000000000000000)
+	testGenesis = &blockchain.Genesis{
+		Config: config,
+		Alloc:  blockchain.GenesisAlloc{address: {Balance: funds}},
+	}
+	genesis = testGenesis.MustCommit(gendb)
+	signer  = types.NewEIP155Signer(config.ChainID)
+)
+
+type SuiteRepository struct {
+	suite.Suite
+	repo *repository
+}
+
+func genRandomAddress() common.Address {
+	key, _ := crypto.GenerateKey()
+	return crypto.PubkeyToAddress(key.PublicKey)
+}
+
+func genRandomHash() (h common.Hash) {
+	hasher := sha3.NewKeccak256()
+
+	r := rand.Uint64()
+	rlp.Encode(hasher, r)
+	hasher.Sum(h[:0])
+
+	return h
+}
+
+func makeChainEventsWithInternalTraces(numBlocks int, genTxs func(i int, block *blockchain.BlockGen)) ([]blockchain.ChainEvent, error) {
+	db := database.NewMemoryDBManager()
+	testGenesis.MustCommit(db)
+
+	// create new blockchain with enabled internal tx tracing option
+	b, _ := blockchain.NewBlockChain(db, nil, testGenesis.Config, gxhash.NewFaker(), vm.Config{Debug: true, EnableInternalTxTracing: true})
+	defer b.Stop()
+
+	// subscribe a new chain event channel
+	chainEventCh := make(chan blockchain.ChainEvent, numBlocks)
+	subscription := b.SubscribeChainEvent(chainEventCh)
+	defer subscription.Unsubscribe()
+
+	// generate blocks
+	blocks, _ := blockchain.GenerateChain(testGenesis.Config, genesis, gxhash.NewFaker(), gendb, numBlocks, genTxs)
+
+	// insert the generated blocks into the test chain
+	if _, err := b.InsertChain(blocks); err != nil {
+		return nil, err
+	}
+
+	var events []blockchain.ChainEvent
+	for i := 0; i < numBlocks; i++ {
+		timer := time.NewTimer(1 * time.Second)
+		select {
+		case <-timer.C:
+			return nil, fmt.Errorf("timeout. too late receive a chain event: %v block", i)
+		case ev := <-chainEventCh:
+			events = append(events, ev)
+		}
+		timer.Stop()
+	}
+
+	return events, nil
+}
+
+func setTestDatabase(t *testing.T, mysql *gorm.DB) {
+	// Drop previous test database if possible.
+	if err := mysql.Exec("DROP DATABASE IF EXISTS test").Error; err != nil {
+		if !strings.Contains(err.Error(), "database doesn't exist") {
+			t.Fatal("Unexpected error happened!", "err", err)
+		}
+	}
+	// Create new test database.
+	if err := mysql.Exec("CREATE DATABASE test DEFAULT CHARACTER SET UTF8").Error; err != nil {
+		t.Fatal("Error while creating test database", "err", err)
+	}
+	// Use test database
+	if err := mysql.Exec("USE test").Error; err != nil {
+		t.Fatal("Error while setting test database", "err", err)
+	}
+
+	// Auto-migrate data model from model.DataModels
+	if err := mysql.AutoMigrate(models...).Error; err != nil {
+		t.Fatal("Error while auto migrating data models", "err", err)
+	}
+}
+
+func (s *SuiteRepository) SetupSuite() {
+	id := "root"
+	password := "root"
+
+	mysql, err := gorm.Open("mysql", fmt.Sprintf("%s:%s@/?parseTime=True", id, password))
+	if err != nil {
+		s.T().Log("Failed connecting to mysql", "id", id, "password", password, "err", err)
+		s.T().Skip()
```

Comment:
Is it ok to skip if there is an failure?

Reply:
Yes, this is in purpose. The test can be running although the database is not on.